docs/reference/scripting/common-script-uses.asciidoc

```diff
@@ -0,0 +1,423 @@
+the associated values. Let's index some a few documents containing log data to use as
+an example:
+
+[source,console]
+----
+POST /my-index/_bulk?refresh
+{"index":{}}
+{"gc": "[2021-04-27T16:16:34.699+0000][82460][gc,heap,exit]   class space    used 266K, capacity 384K, committed 384K, reserved 1048576K"}
+{"index":{}}
+{"gc": "[2021-03-24T20:27:24.184+0000][90239][gc,heap,exit]   class space    used 15255K, capacity 16726K, committed 16844K, reserved 1048576K"}
+{"index":{}}
+{"gc": "[2021-03-24T20:27:24.184+0000][90239][gc,heap,exit]  Metaspace       used 115409K, capacity 119541K, committed 120248K, reserved 1153024K"}
+{"index":{}}
+{"gc": "[2021-04-19T15:03:21.735+0000][84408][gc,heap,exit]   class space    used 14503K, capacity 15894K, committed 15948K, reserved 1048576K"}
+{"index":{}}
+{"gc": "[2021-04-19T15:03:21.735+0000][84408][gc,heap,exit]  Metaspace       used 107719K, capacity 111775K, committed 112724K, reserved 1146880K"}
+{"index":{}}
+{"gc": "[2021-04-27T16:16:34.699+0000][82460][gc,heap,exit]  class space  used 266K, capacity 367K, committed 384K, reserved 1048576K"}
+----
+
+Looking at the data again, there's a timestamp, some other data that you're not
+interested in, and then the `used`, `capacity`, and `committed` data:
+
+[source,txt]
+----
+[2021-04-27T16:16:34.699+0000][82460][gc,heap,exit]   class space    used 266K, capacity 384K, committed 384K, reserved 1048576K
+----
+// NOTCONSOLE
+
+You can assign variables to each part of the data in the `gc` field, and then return
+only the parts that you want. Anything in curly braces `{}` is considered a variable.
+For example, the variables `[%{@timestamp}][%{code}][%{desc}]` will match the first
+three chunks of data, all of which are in square brackets `[]`.
+
+[source,txt]
+----
+[%{@timestamp}][%{code}][%{desc}]  %{ident} used %{usize}, capacity %{csize}, committed %{comsize}, reserved %{rsize}
+----
+// NOTCONSOLE
+
+Your dissect pattern can include the terms `used`, `capacity`, and `committed` instead
+of using variables, because you want to return those terms exactly. You also assign
+variables to the values you want to return, such as `%{usize}`, `%{csize}`, and 
+`%{comsize}`. The separator in the log data is a comma, so your dissect pattern also
+needs to use that separator.
+
+Now that you have a dissect pattern, you can include it in a Painless script as part
+of a runtime field. The script uses your dissect pattern to split apart the `gc`
+field, and then returns exactly the information that you want as defined by the
+`emit` method. Because dissect uses simple syntax, you just need to tell it exactly
+what you want. 
+
+The following pattern tells dissect to return the term `used` followed by a space, plus
+the value from `gc.usize` and a comma. This pattern repeats for the other data that you
+want to retrieve:
+
+[source,txt]
+----
+emit("used" + ' ' + gc.usize + ', ' + "capacity" + ' ' + gc.csize + ', ' + "committed" + ' ' + gc.comsize)
+----
```

Comment:
This is a good thing for debugging but not super useful in production. In production you'll want to just `emit(gc.usize)` or something, right? Worth pointing out.

Reply:
Why is this not useful in production? Is it too slow or just not something that people would typically want to do?

Reply:
Yeah, its somewhat more slow than returning just the number you need, but I think the bigger thing is that folks will typically want to have range queries for the numbers or do math with them or group them with aggs or something - basically if you get numbers I think typically you want to extract them as `long` or `double`. But what you've got it useful to look at, especially because we don't yet have a way to emit all of the extracted values at once. When we have that I think it'd be easier to have folks do *that* and fetch them all in the `fields`, even when they are just looking at things.